src/v/debug_bundle/tests/types_test.cc

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "debug_bundle/types.h"
+#include "model/fundamental.h"
+
+#include <absl/container/flat_hash_map.h>
+#include <gtest/gtest.h>
+
+#include <iomanip>
+
+using namespace debug_bundle;
+
+TEST(types_test, convert_string_to_special_date_success) {
+    const absl::flat_hash_map<ss::sstring, special_date> names = {
+      {"yesterday", special_date::yesterday},
+      {"today", special_date::today},
+      {"now", special_date::now},
+      {"tomorrow", special_date::tomorrow}};
+
+    for (auto&& [k, v] : names) {
+        EXPECT_NO_THROW(EXPECT_EQ(v, boost::lexical_cast<special_date>(k)))
+          << v << " != " << k;
+        EXPECT_EQ(fmt::format("{}", v), k);
+        time_variant tv(v);
+        EXPECT_EQ(fmt::format("{}", tv), k);
+    }
+}
+
+TEST(types_test, convert_string_to_special_date_failure) {
+    EXPECT_THROW(
+      boost::lexical_cast<special_date>("bloop"), std::runtime_error);
+    EXPECT_THROW(
+      boost::lexical_cast<special_date>("YESTERDAY"), std::runtime_error);
+}
+
+TEST(types_test, test_print_iso_8061) {
+    const std::string_view test_time = "2024-09-05T14:34:02";
+    std::istringstream ss(test_time.data());
+    std::tm tmp{};
+    ASSERT_NO_THROW(ss >> std::get_time(&tmp, "%FT%T"));
+    ASSERT_FALSE(ss.fail());
+    std::time_t tt = std::mktime(&tmp);
+    auto tp = debug_bundle_clock::from_time_t(tt);
+    time_variant tv(tp);
+    EXPECT_EQ(fmt::format("{}", tv), test_time);
+}
```

Comment:
just ran this locally because i was curious whether `-c 1` was actually necessary (I don't think it is), and this test failed off by one on the hour. I haven't given it any thought as to why, but FYI.

Reply:
.... I wonder if it's some sort of timezone thing.....

Reply:
> or daylight savings

when are they going to make that illegal!